### DISCUSSION TITLE
Issue with Installing Dependencies in React.js App

### DISCUSSION THREAD

**Discussion by Usman-CrYpToo** [WalletConnect/walletconnect-monorepo#3550] [Developer Support] [answered]
I'm encountering an issue while attempting to install dependencies for my React.js app following the provided documentation. Interestingly, I've noticed that when I attempt to install the same dependencies in a different Next.js app, they install without any issues. I would greatly appreciate any insights or guidance on resolving this discrepancy.

Problem:
When running the command npm install @web3modal/wagmi@3.0.0-alpha.4 wagmi viem for my React.js app, I encounter the following er
![jlkewjrewwr](https://github.com/WalletConnect/walletconnect-monorepo/assets/130186574/0709850f-a45b-4237-95ea-d34e30ca8a87)

...
Error Details:

While resolving: integration@0.1.0
Found: typescript@4.9.5
Conflicting peer dependency: typescript@5.2.2
The error seems to be related to conflicting dependencies, particularly involving TypeScript's version.

Observation:
Interestingly, I tried installing the same dependencies in a separate Next.js app, and they installed without any issues. This suggests that the problem might be specific to my current React.js app setup.

Steps Taken:
I've attempted various troubleshooting steps including clearing the npm cache, verifying compatibility between packages, and updating package versions. Despite these efforts, I have not been able to resolve the issue in my React.js app.

Expected Outcome:
I anticipate being able to install the required dependencies smoothly, just as I have experienced in my Next.js app. The discrepancy between the two apps perplexes me, and I'm eager to understand and overcome this problem.

Request for Help:
I kindly request assistance in identifying the root cause of this discrepancy and finding a solution. As the dependencies are working as expected in my Next.js app, I believe that understanding what's causing the issue in my React.js app would provide valuable insights.

Thank you in advance for your time and expertise. Your support would be immensely helpful in helping me make progress with my project.

**Comment by boidushya** [comment]
Have you tried using `--legacy-peer-deps`?
`npm install @web3modal/wagmi@3.0.0-alpha.4 wagmi viem --legacy-peer-deps`

**Reply by Usman-CrYpToo** [reply]
Yeah, I tried that too, but it just made things worse with more errors. It's a bit frustrating.

**Reply by boidushya** [reply]
Could you share a screenshot of errors you face after --legacy-peer-deps?

**Reply by shkvik** [reply]
@boidushya I have same problem with react

**Reply by boidushya** [reply]
Here's a demo with React and web3modal alpha.4
https://codesandbox.io/p/sandbox/web3modal-v3-vc6t6j?file=/package.json:14,33
Could you make sure the dependencies and versions match up/are correct?

**Reply by Usman-CrYpToo** [reply]
@boidushya After taking your advice, I attempted the installation once again using the --legacy-peer-deps flag. While this approach did allow the dependencies to be installed, I'm now encountering a new set of issues when trying to import the function into my React app.

Upon importing the function, I'm confronted with the following errors and warnings that I had initially reported:

1) - 
![Capture](https://github.com/WalletConnect/walletconnect-monorepo/assets/130186574/cb84f367-eada-4826-bcd9-926d7e0d6aed)
![jhjkhjkhj](https://github.com/WalletConnect/walletconnect-monorepo/assets/130186574/fdbc6bc9-996d-4e95-95e6-f58c546a19fe)
![sjklsjksj](https://github.com/WalletConnect/walletconnect-monorepo/assets/130186574/91c3a124-ed09-4355-8623-004649639ff2)
Module Resolution Error:
An error message appears as follows:
**ERROR in ./node_modules/@web3modal/scaffold/dist/esm/src/client.js 92:52-79
Module not found: Error: Can't resolve './modal/w3m-modal' in 'C:\Users\usman\OneDrive\Desktop\prom\frontend\node_modules\@web3modal\scaffold\dist\esm\src'
Did you mean 'index.js'?
BREAKING CHANGE: The request './modal/w3m-modal' failed to resolve only because it was resolved as fully specified
(probably because the origin is strict EcmaScript Module, e. g. a module with javascript mimetype, a '*.mjs' file, or a '*.js' file where the package.json contains '"type": "module"').
The extension in the request is mandatory for it to be fully specified.
Add the extension to the request.**

2) -Source Map Warnings:
In addition, I continue to observe warnings related to source maps:

**WARNING in ./node_modules/@web3modal/wagmi/dist/esm/src/utils/provider.js
Module Warning (from ./node_modules/source-map-loader/dist/cjs.js):
Failed to parse source map from 'C:\Users\usman\OneDrive\Desktop\prom\frontend\node_modules\@web3modal\wagmi\src\utils\provider.ts' file: Error: 
ENOENT: no such file or directory, open 'C:\Users\usman\OneDrive\Desktop\prom\frontend\node_modules\@web3modal\wagmi\src\utils\provider.ts'**

While the --legacy-peer-deps flag did help with the initial installation problem, it appears to have introduced a new set of complications. I'm eager to hear your insights and suggestions on how I might proceed to address these challenges.

Thank you for your continued support and assistance.

**Reply by boidushya** [reply]
Are you using `create-react-app` by any chance?

**Reply by Usman-CrYpToo** [reply]
yes

**Reply by boidushya** [reply, marked as answer]
`create-react-app` is deprecated and not maintained by the React core team anymore, can you try using Vite instead and see if your issues persist? https://vitejs.dev/guide/
I'd also like to mention that Web3Modal v3 is in alpha right now and even though we appreciate you testing it out, we don't advise you to use it in a production environment.
If you want to use Web3Modal in production, please use v2 instead
https://docs.walletconnect.com/2.0/web3modal/react/wagmi/installation

**Reply by Usman-CrYpToo** [reply]
I'm grateful for letting me know about the deprecation of Create React App. I wasn't aware of this change, and your update is valuable.
I'll explore Vite as an alternative and see if it helps resolve the issues. I'll follow the link you shared to learn more.
Also, thanks for clarifying that Web3Modal v3 is in alpha. I'll consider using v2 for production .

Thanks again for your assistance.

**Comment by zaid-rana** [comment]
Yes i am also facing the same issue

**Comment by samartsevigor** [comment]
What's the solution?
